zeroconf.py

```diff
@@ -1794,13 +1795,19 @@ def new_socket():
 
 
 def get_errno(e):
-    assert isinstance(e, socket.error)
-    return e.args[0]
+    if isinstance(e, (OSError, socket.error,)):
+        return e.errno
+    try:
+        return e.args[0]
+    except IndexError:
+        return None
```

Comment:
It appeared that the `OSError, socket.error` exceptions had an explicit `errno` attribute so I made this match that, and also be defensive since the previous `assert` could be removed with `python -O`.